### PR TITLE
Feature/tooltip on stack bar

### PR DESCRIPTION
**Pull request by cuginoAle**
This PR adds the tooltip to the `stack-bar`.
It also fixes the [bug reported here](https://github.com/nearform/node-clinic-flame/pull/22#issuecomment-437833058)
This PR introduces the new `flame-graph-tooltip-content` component that contains the markup and the logic needed by the `tooltip` when hovering over a frame.
The `stack-bar` component itself has been updated to better support the edge case when the user hovered/clicked on the gap between frames.

screenshot 2018-11-13 at 11 15 11

Something we probably would need to fine tune is the tooltip behaviour when a frame is zoomed in.
Currently we show the tooltip, centre-aligned to the selected frame, on `animation-end`. Maybe we want to rethink this also because we can now zoom a frame in from the stack-bar, which sits in the top section of the page, far away from the zoomed frame and the tooltip itself.

### CODE REVIEW

**Comment by AlanSl** [comment]
Nice. One behavioural thing - this makes it possible for someone to click the 'Expand' button on a frame that is above the top scroll position of the viewport. Could we maybe detect such cases and scroll up to show say 20% above the expanded node on zoom? Otherwise it's a bit confusing as to what is happening, everything just goes black.

General tip - can we keep PRs to one thing at a time. Even if that means a PR is tiny - tiny PRs are good, it's better to have a micro-PR for the fix for the copy paths button bug, as we can merge it straight away and that bug fix doesn't get delayed by discussion and review of other features.

**Comment by cuginoAle** [comment]
@AlanSl You are going to like this.
When a node is selected the canvas scrolls it into view (only if not already visible)
Also, the user can now dblClick on the stack-bar to zoom in/out.

**Comment by cuginoAle** [comment]
Any suggestion on how to fix those CI errors?

File: visualizer/flame-graph-tooltip-content.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+'use strict'
+const d3 = require('d3')
```
Comment (goto-bus-stop) [inline visualizer/flame-graph-tooltip-content.js]: since #26 we no longer depend on the full d3, it should work if you use `./d3.js` instead

**Comment by cuginoAle** [comment]
> (I kinda feel like this functionality might work better as [buttons in the frame info area](https://trello.com/c/bUbKQEHM/514-show-expand-copy-buttons-in-the-frame-info-at-the-top) though, so you'd click a frame in the stack bar and then access the expand/copy buttons from there…)

Yes... we might not need the tooltip in the stack-bar after all. The user can dblClick on the stack-bar to zoom in/out and select/copy the path from the info-box. Will create a new PR and test this out

**Pull request merged**